compiler/syntax/src/res_core.ml

```diff
@@ -2785,26 +2742,22 @@ and parse_jsx_prop p : Parsetree.jsx_prop option =
         Parser.next p;
         (* no punning *)
         let optional = Parser.optional p Question in
-        Scanner.pop_mode p.scanner Jsx;
         let attr_expr = parse_primary_expr ~operand:(parse_atomic_expr p) p in
         Some (Parsetree.JSXPropValue ({txt = name; loc}, optional, attr_expr))
       | _ -> Some (Parsetree.JSXPropPunning (false, {txt = name; loc})))
   (* {...props} *)
   | Lbrace -> (
-    Scanner.pop_mode p.scanner Jsx;
```

Comment:
This is rather confusing when you are in a nested jsx scenario:
```res
<div>
  <p>
    {foo}
  </p>
</div>
```
Popping `Jsx` from `p` requires the pop of `div` also to happen to get out of `Jsx` mode.